### PR TITLE
[Feature] CLI Unit Testing

### PR DESCRIPTION
**Pull request by antonio-amjr**
Fix: https://github.com/project-chip/certification-tool/issues/726
---

# Description
This change adds the testing infrastructure, unit tests, test coverage and testing documentation with the following:

## Test Architecture 🏗️

✅ Naming Conventions

- **Test files**: `test_<module_name>.py`
- **Test classes**: `Test<ClassName>`
- **Test methods**: `test_<function_name>_<scenario>_<expected_outcome>`
- **Fixtures**: Descriptive names following snake_case

✅ Documentation Standards

- Comprehensive docstrings explaining test purpose
- Inline comments for complex test logic
- README documentation for test usage

## Test Categories & Coverage 🧪
1. **Command-Level Tests (CLI Interface)**
Each CLI command has dedicated test coverage:

| Command | Test File | Test Methods | Coverage Areas |
|---------|-----------|--------------|----------------|
| `abort-testing` | `test_abort_testing.py` | 19 | Success, API errors, client cleanup |
| `available-tests` | `test_available_tests.py` | 18 | JSON/YAML output, empty responses, error handling |
| `project` | `test_project_commands.py` | 24 | Config validation, file handling, API integration, Confirmation prompts, API errors, validation, pagination, filtering, output formats, config updates, validation, error handling |
| `run-tests` | `test_run_tests.py` | 31 | Complex async workflows, WebSocket handling |
| `test-run-execution` | `test_test_run_execution.py` | 48 | Pagination, filtering, state handling, log content handling, large outputs |
| `test-runner-status` | `test_test_runner_status.py` | 16 | State visualization, JSON output |

2. **Utility Function Tests**

- **File**: test_utils.py (24 test methods)
- **Configuration Processing**: Properties files, PICS XML parsing
- **Data Transformation**: Test selection building, config merging
- **Error Handling**: Malformed inputs, missing files
- **Edge Cases**: Circular references, empty data structures

### Shared Fixtures (conftest.py)
Comprehensive fixture library providing:

**CLI Testing**: Click test runners with proper isolation
**Mock Infrastructure**: API clients (sync/async), external services
**Test Data**: Sample projects, test collections, configurations
**File Management**: Temporary directories, mock config files
**Environment**: Color management, consistent test environment

### Configuration (pyproject.toml)
Pytest configuration:
```toml
[tool.pytest.ini_options]
minversion = "7.0"
addopts = [
    "--strict-markers",
    "--strict-config", 
    "--cov=th_cli",
    "--cov-report=term-missing",
    "--cov-report=html:htmlcov",
    "--cov-report=xml:coverage.xml",
    "--cov-branch",
    "--cov-fail-under=85"
]
```

### 1. Test Isolation & Independence

- **No External Dependencies**: All API calls mocked
- **Temporary File Management**: Clean setup/teardown
- **Independent Test Execution**: Tests can run in any order
- **State Isolation**: No shared state between tests

### 2. Comprehensive Error Testing

Every command includes tests for:
- ✅ API error responses (400, 401, 403, 404, 500, 503)
- ✅ Network connectivity issues  
- ✅ Configuration errors
- ✅ Invalid input validation
- ✅ File system errors
- ✅ Malformed response handling

### 3. Modern Testing Patterns

- **Parameterized Tests**: Reduce duplication for similar scenarios
- **Fixture Dependency Injection**: Clean, reusable test setup
- **Mock Verification**: Ensure correct API interactions
- **Async Testing**: Proper handling of async CLI commands
- **Context Managers**: Resource cleanup and management

### 4. Security Testing

- **Input Sanitization**: Path traversal prevention
- **Configuration Validation**: Secure defaults
- **Error Information Leakage**: Safe error messages

### CODE REVIEW

**Review by hiltonlima — APPROVED** [review]

File: th_cli/validation.py [hunk 1]
```diff
@@ -49,8 +48,8 @@ def _sanitize_path(file_path: str) -> Path:
         # This will raise ValueError if path tries to go outside current working directory
         path.relative_to(Path.cwd().resolve())
     except ValueError:
-        # Allow absolute paths in user directories but fail paths outside
-        if not str(path).startswith(str(Path.home())):
+        # Allow absolute paths in user directories or system tmp, but fail paths outside
+        if not (str(path).startswith(str(Path.home())) or str(path).startswith(str(Path("/tmp")))):
```
[the remark this answers is not in the document]
Comment (antonio-amjr) [reply th_cli/validation.py:52]: This will be managed in the future along with some extra unit tests. The infrastructure here is more important to enter right away.
